### PR TITLE
Cleanup of caikit OOTB

### LINKED ISSUE #2143: [Bug]: Cleanup caikit ootb runtime
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

Installing Dashboard directly (eg. `v2.12.0`, `commit xyz`, `branch name`)

### Version

main

### Current Behavior

A little bit of a cleanup still needed for the [caikit-ootb.yaml](https://github.com/opendatahub-io/odh-dashboard/blob/8344e6d3761e3def23c2357f01dd4942870cf452/manifests/modelserving/caikit-ootb.yaml).

### Expected Behavior

Add an additional env var in [caikit-ootb.yaml](https://github.com/opendatahub-io/odh-dashboard/blob/8344e6d3761e3def23c2357f01dd4942870cf452/manifests/modelserving/caikit-ootb.yaml#L42C1-L44C31)

Expected output:
```
        env:
          - name: RUNTIME_LOCAL_MODELS_DIR
            value: /mnt/models
          - name: RUNTIME_GRPC_SERVER_THREAD_POOL_SIZE
            value: 64
```

---

Remove ports name field in [caikit-ootb.yaml](https://github.com/opendatahub-io/odh-dashboard/blob/8344e6d3761e3def23c2357f01dd4942870cf452/manifests/modelserving/caikit-ootb.yaml#L47) as this will cause some issues with KServe.

### Steps To Reproduce

Go to Settings => Serving Runtimes & view the YAML for the ootb caikit runtime

### Workaround (if any)

_No response_

### What browsers are you seeing the problem on?

_No response_

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Closes #2143 

## How Has This Been Tested?

1. Deploy the new template with `oc apply -k manifests/modelserving`
2. Create a new Caikit deployment with kserve
3. Check that you can infer the results

## Test Impact

No testing involved

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by lucferbux** [comment]
@andrewballantyne @Xaenalt This is the cleanup PR for caikit. The renaming of the file is not needed, since it's an internal file and there's no implication whatsoever to the final product. I can do it if you want but I don't see the need.

File: manifests/modelserving/caikit-ootb.yaml [hunk 1]
```diff
@@ -42,8 +42,9 @@ objects:
         env:
           - name: RUNTIME_LOCAL_MODELS_DIR
             value: /mnt/models
+          - name: RUNTIME_GRPC_SERVER_THREAD_POOL_SIZE
+            value: "64"
```
Comment (lucferbux) [inline manifests/modelserving/caikit-ootb.yaml:46]: When deploying, it seems that we cannot have ints as env values, is this alright? @Xaenalt @andrewballantyne

Comment (andrewballantyne) [reply manifests/modelserving/caikit-ootb.yaml:46]: I never tested it on cluster, it was all a slack conversation... if you can't use ints, not sure you can do anything else -- so as long as it works out that's all that matters. @Xaenalt needs to make that call.

**Comment by bdattoma** [comment]
maybe we could add the version metadata as well, otherwise user will see this in the UI:
image

I would say we could review the need of `(name - version)` in the UI since the metadata is not mandatory.

File: manifests/modelserving/caikit-ootb.yaml (hunk 1 above)
Comment (Xaenalt) [reply manifests/modelserving/caikit-ootb.yaml:46]: Yeah, that's fine, shell interprets them the same

Comment (lucferbux) [reply manifests/modelserving/caikit-ootb.yaml:46]: But is that an statement or it's just a guess?

Comment (Xaenalt) [reply manifests/modelserving/caikit-ootb.yaml:46]: Statement

Comment (lucferbux) [reply manifests/modelserving/caikit-ootb.yaml:46]: Great thanks!

**Comment by lucferbux** [comment]
> maybe we could add the version metadata as well, otherwise user will see this in the UI: image
> 
> @Xaenalt does the runtime have a specific version?
> 
> I would say we could review the need of `(name - version)` in the UI since the metadata is not mandatory.

We can add that as a follow up!

**Pull request merged**